packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,200 @@
+ */
+export default function HelpButton({
+  accessibilityPopoverLabel,
+  idealDirection = 'down',
+  link,
+  onClick,
+  text,
+  zIndex,
+}: Props): Node {
+  const ref = useRef(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [isHovered, setHovered] = useState(false);
+  const [isFocused, setFocused] = useState(false);
+  const [isActive, setActive] = useState(false);
+  const popoverId = useId();
+  const {
+    accessibilityTooltipMessage,
+    accessibilityLinkLabel: accessibilityDefaultLinkLabel,
+    accessibilityIcon,
+  } = useDefaultLabelContext('HelpButton');
+
+  useEffect(() => {
+    if (isOpen) {
+      ref?.current?.focus();
+    }
+  }, [isOpen]);
+
+  const toggleView = () => {
+    setIsOpen((currVal) => !currVal);
+    setActive((currVal) => !currVal);
+  };
+
+  const onHandleTap = (props) => {
+    toggleView();
+    if (onClick) {
+      onClick(props);
+    }
+  };
+
+  const bgIconColor = isHovered || isActive || isFocused ? 'default' : 'subtle';
+
+  const tooltipZIndex = zIndex ?? new FixedZIndex(1);
+
+  const zIndexLayer = new CompositeZIndex([new FixedZIndex(tooltipZIndex.index())]);
+
+  const textElement = typeof text === 'string' ? <Text align="center">{text}</Text> : text;
+
+  return (
+    <Flex alignItems="center" justifyContent="center" flex="none">
+      <Tooltip
+        text={accessibilityTooltipMessage}
+        idealDirection={idealDirection}
+        zIndex={tooltipZIndex}
+      >
+        <TapArea
+          accessibilityExpanded={isOpen}
+          accessibilityControls={popoverId}
+          accessibilityLabel={accessibilityTooltipMessage}
+          fullWidth={false}
+          onTap={onHandleTap}
+          ref={ref}
+          role="button"
+          rounding="circle"
+          onBlur={() => {
+            setFocused(false);
+            setActive(false);
+          }}
+          onFocus={() => setFocused(true)}
+          onMouseDown={() => setActive(true)}
+          onMouseEnter={() => setHovered(true)}
+          onMouseLeave={() => setHovered(false)}
+          onMouseUp={() => setActive(false)}
+          tabIndex={0}
+        >
+          <Box
+            width={16}
+            height={16}
+            rounding="circle"
+            display="flex"
+            alignItems="center"
+            justifyContent="center"
+            dangerouslySetInlineStyle={{
+              __style: {
+                backgroundColor: `var(--color-text-${bgIconColor})`,
+              },
+            }}
+          >
+            <Icon
+              accessibilityLabel={accessibilityIcon}
+              color="inverse"
+              icon="question-mark"
+              size={8}
+            />
+          </Box>
+        </TapArea>
+      </Tooltip>
+      {isOpen && (
+        <Box zIndex={zIndexLayer}>
```

Comment:
zIndex not zIndexLayer, the Layer part is confusing

Reply:
I will set as `zIndexWrapper`, because we have a `zIndex` prop already. 🙏🏾

Reply:
DONE